src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,308 @@ object PropertyChecker {
+    (value, _, _) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def columnReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, _, _) => {
+      value match {
+        case s: TextNode => {
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s)
+          (arrayNode, Array[String](), csvwPropertyType)
+        }
+        case a: ArrayNode => (a, Array[String](), csvwPropertyType)
+      }
+    }
+  }
+
+  def targetFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, _, _) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def scriptFormatProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, _, _) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def sourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, _, _) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def resourceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, _, _) => {
+      (value, Array[String](), csvwPropertyType)
+    }
+  }
+
+  def schemaReferenceProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, _) => {
+      val url = baseUrl + value.asText()
+      // Don't know how to place a URI object in JsonNode, keeping the text value as of now
+      (new TextNode(url), Array[String](), csvwPropertyType)
+    }
+  }
+
+  def dialectProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case o: ObjectNode => {
+          val valueCopy = o.deepCopy()
+          var warnings = Array[String]()
+          val fieldsAndValues = Array.from(valueCopy.fields.asScala)
+          for (fieldAndValue <- fieldsAndValues) {
+            val key = fieldAndValue.getKey
+            val v = fieldAndValue.getValue
+            key match {
+              case "@id" => {
+                val matcher = PropertyChecker.startsWithUnderscore.pattern.matcher(v.asText())
+                if (matcher.matches) {
+                  throw new MetadataError("@id starts with _:")
+                }
+              }
+              case "@type" => {
+                if (v.asText() != "Dialect") {
+                  throw new MetadataError("@type of dialect is not 'Dialect'")
+                }
+              }
+              case _ => {
+                val (new_v, w, t) = checkProperty(key, v, baseUrl, lang)
```

Comment:
`new_v` doesn't follow the typical [Scala variable naming conventions](https://docs.scala-lang.org/style/naming-conventions.html) does it?

Reply:
No it doesn't. I have changed it now.